src/udev/udevadm-info.c

```diff
@@ -316,11 +321,19 @@ static int query_device(QueryType query, sd_device* device) {
         case QUERY_PROPERTY: {
                 const char *key, *value;
 
-                FOREACH_DEVICE_PROPERTY(device, key, value)
+                FOREACH_DEVICE_PROPERTY(device, key, value) {
+                        if (arg_properties && !strv_contains(arg_properties, key))
+                                continue;
                         if (arg_export)
                                 printf("%s%s='%s'\n", strempty(arg_export_prefix), key, value);
-                        else
-                                printf("%s=%s\n", key, value);
+                        else {
+                                if (arg_value)
```

Comment:
merge the `else {` and the `if (arg_value)` into `else if (arg_value)`. Saves you one level of indentations.